src/webfaf/reports.py

```diff
@@ -830,13 +830,20 @@ def new():
             osr_id = None
             osr = None
             if report["os"]["name"] in systems:
-                osr = (db.session.query(OpSysRelease)
-                       .join(OpSys)
-                       .filter(OpSys.name ==
-                               systems[report["os"]["name"]].nice_name)
-                       .filter(OpSysRelease.version ==
-                               report["os"]["version"])
-                       .first())
+                try:
+                    osr = (db.session.query(OpSysRelease)
+                           .join(OpSys)
+                           .filter(OpSys.name ==
+                                   systems[report["os"]["name"]].nice_name)
+                           .filter(OpSysRelease.version ==
+                                   report["os"]["version"])
+                           .first())
+                except OperationalError as e:
```

Comment:
The exception can also be a `DatabaseError` or an `InterfaceError`.

Reply:
> Can you make a separate pull-request for the `Makefile` changes?

Sorry, here: https://github.com/abrt/faf/pull/784

Reply:
I added the `try` - `except` blocks in the two more places that you suggested but there seems to be a lot more DB communication throughout this file alone. I wonder if those should be dealt with as well, since, like you told me, you suggested these two particular weak spots based on your logs alone, so there's no telling where else it can possibly happen.